### PR TITLE
osd: fix reqid assignment for reply messages in OpRequest

### PR DESCRIPTION
**Pull request by cyx1231st**
Signed-off-by: Yingxin Cheng <yingxin.cheng@intel.com>

### CODE REVIEW

File: src/osd/OpRequest.cc [hunk 1]
```diff
@@ -38,6 +39,8 @@ OpRequest::OpRequest(Message *req, OpTracker *tracker) :
     reqid = static_cast<MOSDSubOp*>(req)->reqid;
   } else if (req->get_type() == MSG_OSD_REPOP) {
     reqid = static_cast<MOSDRepOp*>(req)->reqid;
+  } else if (req->get_type() == MSG_OSD_REPOPREPLY) {
+    reqid = static_cast<MOSDRepOpReply*>(req)->reqid;
```
Comment (joscollin) [inline src/osd/OpRequest.cc]: What about `MOSDSubOpReply` ?

Comment (joscollin) [inline src/osd/OpRequest.cc]: It would have been a better design if `reqid` was private as it is in `MOSDOp`. Don't know why it is not.

**Comment by cyx1231st** [comment]
retest this please

File: src/osd/OpRequest.cc (hunk 1 above)
Comment (cyx1231st) [inline src/osd/OpRequest.cc]: Done, thanks.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2017-08-20_01:49:41-rados-wip-kefu-testing-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2017-08-19_02:32:27-rados-wip-kefu-testing-distro-basic-mira/

**Pull request merged**